### PR TITLE
Add cron job to clean outdated disks

### PR DESCRIPTION
**Pull request by zhenghuiwang**
Fix
https://github.com/kubeflow/kubeflow/issues/2401

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/311)

### CODE REVIEW

**Comment by zhenghuiwang** [comment]
unmatched and expired disks:

```
Unmatched disks:
devstatsdb-data
gke-kf-v0-4-n00-27fc1b-pvc-739385a7-2352-11e9-a669-42010a8e004e
gke-kf-v0-4-n00-30d4c0-pvc-5f11564c-2bd6-11e9-bdfc-42010a8e0080
gke-kf-v0-4-n00-4e1bce-pvc-83ca10ed-20e6-11e9-a4c1-42010a8e00bc
gke-kf-v0-4-n00-4e1bce-pvc-85216a9d-20e6-11e9-a4c1-42010a8e00bc
...
gke-kfctl-0147-9ea7f0d-pvc-1e96be69-2abe-11e9-8bf9-42010a8e003e
gke-kfctl-0147-9ea7f0d-pvc-1ea5bb0d-2abe-11e9-8bf9-42010a8e003e
gke-kfctl-0147-9ea7f0d-pvc-1ef5ab73-2abe-11e9-8bf9-42010a8e003e
gke-kfctl-0d50-ffd8dfd-pvc-a633fd5b-1f8b-11e9-99c2-42010af0000d
......
gke-kfctl-dda4-a53a944-pvc-794f935a-d7eb-11e8-bbbc-42010a8e0116
gke-kfctl-e12f-cd5774b-pvc-df666a35-2055-11e9-a3c1-42010a8e00ae
gke-kfctl-e496-8b214a1-pvc-cdde0430-20ba-11e9-9427-42010a8e0073
gke-kfctl-eec0-f68bb1a-pvc-160144a6-21ba-11e9-ac07-42010a8e008c
gke-kfctl-f5fe-06bb3ee-pvc-7e2af915-20ec-11e9-a41b-42010a8e00cc
gke-kfctl-fe34-df70357-pvc-deea212e-fd7c-11e8-9a89-42010af000a1
gke-kfctl-fe34-df70357-pvc-df08aaa0-fd7c-11e8-9a89-42010af000a1
gke-kfctl-fe34-df70357-pvc-e7f13e0a-fd7c-11e8-9a89-42010af000a1
gke-kubeflow-testing-default-pool-90dc0402-6q9b
gke-kubeflow-testing-default-pool-90dc0402-8f92
gke-kubeflow-testing-default-pool-90dc0402-b3xn
gke-kubeflow-testing-default-pool-90dc0402-bj6f
gke-kubeflow-testing-default-pool-90dc0402-dzvl
gke-kubeflow-testing-default-pool-90dc0402-g0dg
gke-kubeflow-testing-default-pool-90dc0402-q60c
gke-kubeflow-testing-default-pool-90dc0402-t7n7
gke-kubeflow-testing-default-pool-90dc0402-x2pr
gke-kubeflow-testing-default-pool-90dc0402-z4v3
gke-kubeflow-testing-gpu-pool-117658d4-5qm1
gke-kubeflow-testing-gpu-pool-117658d4-p0kf
gke-kubeflow-testing-gpu-pool-117658d4-pggv
gke-ztor-postsubmit-v1be-default-pool-89e5c5ac-1v20
grafana-data
influxdb-data
kf-v0-4-n00-storage-pipeline-db
kf-v0-4-n00-storage-pipeline-nfs
kf-v0-4-n01-storage-pipeline-db
kf-v0-4-n01-storage-pipeline-nfs
kf-v0-4-n02-storage-pipeline-db
kf-v0-4-n02-storage-pipeline-nfs
kf-v0-4-n03-storage-pipeline-db
kf-v0-4-n03-storage-pipeline-nfs
kf-v0-4-n04-storage-pipeline-db
kf-v0-4-n04-storage-pipeline-nfs
```


```
expired disks:
gke-e2e-0383-9b58dca4--pvc-90a76d97-a576-11e8-99fb-42010a8e01b7
gke-e2e-0383-v1a2-9711-pvc-91948d78-a576-11e8-b3f9-42010a8e0193
gke-e2e-0c91-f555e7c8--pvc-05f74281-a24b-11e8-aa81-42010af00011
gke-e2e-0c91-v1a2-094c-pvc-11f5830d-a24b-11e8-80ff-42010a8e010a
gke-e2e-0cbc-180666e2--pvc-b63000e2-a24b-11e8-a6f4-42010a8e00c8
gke-e2e-0cbc-v1a2-895b-pvc-b50f030b-a24b-11e8-912c-42010a8e0156
gke-e2e-114f-1878704b--pvc-f14b5689-c2a2-11e8-a480-42010af0005d
......
gke-e2e-114f-v1a2-dfc2-pvc-ef715c69-c2a2-11e8-9604-42010a8e0067
gke-e2e-ee09-v1a2-9e22-pvc-14d35c04-cf49-11e8-b5fa-42010a8e0096
gke-e2e-ef23-47d286be--pvc-61e61b4c-c277-11e8-a4f8-42010a8e00e2
gke-e2e-ef23-v1a2-81d2-pvc-620eb468-c277-11e8-93f9-42010af00005
kubeflow-postsubmit-kubeflow-e2e-minikube-c1d65f6-696-9ea0
zeflow-presubmit-kubeflow-e2e-minikube-1224-a45247a-2608-5d81
zeflow-presubmit-kubeflow-e2e-minikube-1224-e8f6cef-2600-fc0f
zeflow-presubmit-kubeflow-e2e-minikube-1224-ef163c8-2611-6f23
zeflow-presubmit-kubeflow-e2e-minikube-1638-7cefc39-3555-8823
```

**Comment by zhenghuiwang** [comment]
sample of the disks **to be deleted**

```
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-1e7c-f8468df4--pvc-f46aedac-c2bd-11e8-b959-42010a8e0050, age = datetime.timedelta(138, 1813, 93142
0)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-2305-e406d4fc--pvc-8af0ab98-a58f-11e8-ba01-42010a8e0060, age = datetime.timedelta(175, 13531, 3754
83)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-2305-v1a2-96b0-pvc-7a933417-a58f-11e8-9740-42010a8e0032, age = datetime.timedelta(175, 13558, 7525
34)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-244b-v1a2-d4a1-pvc-b2c23aab-b09c-11e8-8ed4-42010a8e01f0, age = datetime.timedelta(161, 8018, 25458
3)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-2773-b125f94f--pvc-bde01c31-cc45-11e8-be59-42010af00087, age = datetime.timedelta(125, 77133, 4456
33)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-2773-v1a2-25ae-pvc-b6e79782-cc45-11e8-8eff-42010a8e009a, age = datetime.timedelta(125, 77145, 1266
82)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-31f0-773e1696--pvc-7400dbc2-a58e-11e8-9323-42010a8e0089, age = datetime.timedelta(175, 13999, 3717
32)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-31f0-v1a2-4bc7-pvc-6fdeb3e3-a58e-11e8-a8eb-42010af0009c, age = datetime.timedelta(175, 14006, 2677
82)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-3848-a0e3f3d4--pvc-2980d3e5-cc43-11e8-a34b-42010af00009, age = datetime.timedelta(125, 78241, 5483
1)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-3848-v1a2-1506-pvc-17a579f4-cc43-11e8-a1e5-42010a8e0043, age = datetime.timedelta(125, 78271, 3418
80)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-3a0e-e337392d--pvc-674a16ce-cc06-11e8-9f49-42010a8e0113, age = datetime.timedelta(126, 17937, 3893
0)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-3a0e-v1a2-c9c6-pvc-4bf6f7fd-cc06-11e8-bcff-42010a8e0124, age = datetime.timedelta(126, 17982, 8319
79)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-45ac-60282ca7--pvc-ece91365-cb70-11e8-ad6e-42010a8e0021, age = datetime.timedelta(126, 82137, 2630
29)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-45ac-v1a2-ae39-pvc-ddaecb6b-cb70-11e8-a7af-42010a8e002c, age = datetime.timedelta(126, 82162, 9360
78)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-5ef1-a7aad7b2--pvc-80f55467-a588-11e8-96f4-42010a8e0063, age = datetime.timedelta(175, 16554, 5571
27)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-655f-c741a4b6--pvc-27451f15-cf4e-11e8-a858-42010a8e001a, age = datetime.timedelta(122, 2867, 13917
7)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-69e1-v1a2-7d9b-pvc-4e6b6393-b090-11e8-9f1e-42010a8e0104, age = datetime.timedelta(161, 13340, 4382
26)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-6c15-1dff903d--pvc-f8ca95af-cb22-11e8-8be8-42010a8e0075, age = datetime.timedelta(127, 29218, 1442
76)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-729b-c643190d--pvc-6fb53d1f-cb52-11e8-b831-42010a8e00ab, age = datetime.timedelta(127, 8832, 20432
5)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-7f97-v1a2-8169-pvc-4313ce73-b09c-11e8-af26-42010af00231, age = datetime.timedelta(161, 8205, 46937
4)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-818e-99c50c43--pvc-51fc8251-c299-11e8-b3c0-42010a8e0122, age = datetime.timedelta(138, 17548, 3474
23)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-818e-v1a2-4e47-pvc-4b0f64db-c299-11e8-a87d-42010af00163, age = datetime.timedelta(138, 17559, 7554
72)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-8560-50ef0d91--pvc-01c478ef-c2a0-11e8-a7e5-42010a8e006f, age = datetime.timedelta(138, 14676, 3225
22)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-8560-v1a2-e110-pvc-17650971-c2a0-11e8-a001-42010a8e0074, age = datetime.timedelta(138, 14639, 9465
76)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-8e71-v1a2-7874-pvc-5eeee717-a00b-11e8-9f86-42010a8e00c5, age = datetime.timedelta(182, 15254, 5566
26)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-95af-83511a5d--pvc-8c2cc0db-afbe-11e8-85c4-42010af0022a, age = datetime.timedelta(162, 17031, 3606
76)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-95af-v1a2-e7b2-pvc-832d6585-afbe-11e8-ab38-42010a8e01d8, age = datetime.timedelta(162, 17046, 3617

85)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-dc4b-21b77693--pvc-572f6f82-ccb2-11e8-b18c-42010af00055, age = datetime.timedelta(125, 30490, 5574
34)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-dc4b-v1a2-1392-pvc-50289505-ccb2-11e8-923a-42010af00015, age = datetime.timedelta(125, 30502, 3414
83)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-dcf4-88e1c8fe--pvc-d193d81e-cbee-11e8-a0e4-42010af00094, age = datetime.timedelta(126, 28066, 6505
32)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-dcf4-v1a2-a399-pvc-bec576df-cbee-11e8-9179-42010a8e0034, age = datetime.timedelta(126, 28098, 1505
82)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-e28d-v1a2-46ba-pvc-c9b83d7e-8939-11e8-b441-42010a8e009d, age = datetime.timedelta(211, 18595, 1563
1)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-e328-e0cbcdd5--pvc-14a7017b-a571-11e8-b54b-42010a8e001c, age = datetime.timedelta(175, 26614, 6956
87)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-e328-v1a2-5717-pvc-1bcc6c4d-a571-11e8-8f60-42010a8e000c, age = datetime.timedelta(175, 26602, 6877
36)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-e840-ca22bcbc--pvc-cd4431b3-c29c-11e8-97d7-42010a8e0050, age = datetime.timedelta(138, 16052, 8837
86)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-ec7b-7660f17f--pvc-c59ee754-a4e1-11e8-a37a-42010a8e0017, age = datetime.timedelta(176, 1765, 19183
5)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-ec7b-v1a2-6602-pvc-b8027959-a4e1-11e8-9e3b-42010a8e0111, age = datetime.timedelta(176, 1788, 20288
5)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-ee09-3fad19e6--pvc-077f7c30-cf49-11e8-966e-42010a8e0102, age = datetime.timedelta(122, 5067, 94793
4)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-ee09-v1a2-9e22-pvc-14d35c04-cf49-11e8-b5fa-42010a8e0096, age = datetime.timedelta(122, 5045, 55498
4)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-ef23-47d286be--pvc-61e61b4c-c277-11e8-a4f8-42010a8e00e2, age = datetime.timedelta(138, 32124, 5040
34)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: gke-e2e-ef23-v1a2-81d2-pvc-620eb468-c277-11e8-93f9-42010af00005, age = datetime.timedelta(138, 32124, 2400
83)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: kubeflow-postsubmit-kubeflow-e2e-minikube-c1d65f6-696-9ea0, age = datetime.timedelta(144, 19486, 672315)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: zeflow-presubmit-kubeflow-e2e-minikube-1224-a45247a-2608-5d81, age = datetime.timedelta(210, 31928, 801377
)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: zeflow-presubmit-kubeflow-e2e-minikube-1224-e8f6cef-2600-fc0f, age = datetime.timedelta(211, 11728, 95429)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: zeflow-presubmit-kubeflow-e2e-minikube-1224-ef163c8-2611-6f23, age = datetime.timedelta(210, 28260, 631480
)
INFO|2019-02-12T17:59:48|py/kubeflow/testing/cleanup_ci.py|157| Deleting disk: zeflow-presubmit-kubeflow-e2e-minikube-1638-7cefc39-3555-8823, age = datetime.timedelta(138, 82996, 21530)

```

**Comment by jlewi** [comment]
I don't think you want to delete the disks `gke.*`

```gke-e2e-dc4b-v1a2-1392-pvc-50289505-ccb2-11e8-923a-42010af00015```

These disks belong to GKE nodes.

**Comment by zhenghuiwang** [comment]
> I don't think you want to delete the disks `gke.*`
> 
> `gke-e2e-dc4b-v1a2-1392-pvc-50289505-ccb2-11e8-923a-42010af00015`
> 
> These disks belong to GKE nodes.

I see. Excluding `gke*` from [the full list of disk names](https://developers.google.com/apis-explorer/#p/compute/v1/compute.disks.list?project=kubeflow-ci&zone=us-east1-d&fields=items(name%252CsizeGb)%252CnextPageToken&_h=2&), I think we only need to delete ones of name`*presubmit*` and `*postsubmit*`.  Updated the matching criteria to that. 

That is 500GB for
```
kubeflow-postsubmit-kubeflow-e2e-minikube-c1d65f6-696-9ea0, age = datetime.timedelta(144, 41282, 656660)
zeflow-presubmit-kubeflow-e2e-minikube-1224-a45247a-2608-5d81, age = datetime.timedelta(210, 53724, 785959
)
zeflow-presubmit-kubeflow-e2e-minikube-1224-e8f6cef-2600-fc0f, age = datetime.timedelta(211, 33524, 80095)
zeflow-presubmit-kubeflow-e2e-minikube-1224-ef163c8-2611-6f23, age = datetime.timedelta(210, 50056, 616209
)
zeflow-presubmit-kubeflow-e2e-minikube-1638-7cefc39-3555-8823, age = datetime.timedelta(139, 18392, 6317)
```

**Comment by kunmingg** [comment]
Can we also clean deployments in kubeflow-ci which has name suffix ```storage``` and life > 1 day?

**Comment by zhenghuiwang** [comment]
> Can we also clean deployments in kubeflow-ci which has name suffix `storage` and life > 1 day?

Can you give some examples? I can create a separate PR to clean the `storage` deployments

**Comment by kunmingg** [comment]
Found where the leak comes from: https://github.com/kubeflow/kubeflow/issues/2465#issuecomment-463418220
We don't need to worry about it in this PR then

File: py/kubeflow/testing/cleanup_ci.py [hunk 1]
```diff
@@ -117,6 +127,47 @@ def cleanup_endpoints(args):
...
+      for d in results["items"]:
+        name = d["name"]
+        if not is_match_disk(name):
+          unmatched.append(name)
+          continue
+
+        age = getAge(d["creationTimestamp"])
+        if age > datetime.timedelta(hours=args.max_age_hours):
+          logging.info("Deleting disk: %s, age = %r", name, age)
+          disks.delete(project=args.project, zone=zone, disk=name)
```
Comment (kunmingg) [inline py/kubeflow/testing/cleanup_ci.py]: Will disks.delete throw exception when disk can not be deleted? 
Like still used by a pod?
We might want to catch and continue here.

Comment (zhenghuiwang) [reply py/kubeflow/testing/cleanup_ci.py]: there won't be exception thrown. but I updated the code to print out the response.

https://developers.google.com/resources/api-libraries/documentation/compute/v1/python/latest/compute_v1.disks.html#delete

**Comment by jlewi** [comment]
/lgtm
/approve
/hold
@kunmingg Any additional comments?

**Comment by kunmingg** [comment]
No more, lgtm, thanks!
/hold cancel

**Pull request merged**